extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/rules/controllers.rules.js

```diff
@@ -359,25 +359,26 @@ angular.module('PaperUI.controllers.rules', [ 'PaperUI.controllers.extension' ])
             });
 
             var index = sharedProperties.searchArray(sharedProperties.getModuleArray(type), $scope.id);
-            if (index != -1) {
+            if (index !== -1) {
                 $scope.configuration = configService.convertValues(sharedProperties.getModuleArray(type)[index].configuration);
                 angular.copy($scope.configuration, originalConfiguration);
             }
             $scope.configuration = configService.setConfigDefaults($scope.configuration, $scope.parameters);
-            if (hasScript && type != 'trigger') {
+            if (hasScript && type !== 'trigger') {
                 var triggers = sharedProperties.getModuleArray('trigger');
                 angular.forEach(triggers, function(trigger, i) {
                     var moduleType = filterByUid($scope.triggerData, trigger.type);
-                    if (moduleType && moduleType.length > 0) {
+                    if (moduleType && moduleType.length) {
                         $scope.items = $scope.items.concat(moduleType[0].outputs);
                     }
                 });
                 if (type == 'action') {
                     var actions = sharedProperties.getModuleArray('action');
-                    for (var i = 0; i < sharedProperties.searchArray(actions, $scope.id); i++) {
-                        var moduleType = filterByUid($scope.moduleData, actions[i].type);
-                        if (moduleType[0] && moduleType[0].outputs && moduleType[0].outputs.length > 0) {
-                            $scope.items = $scope.items.concat(moduleType[0].outputs);
+                    var actionIndex = sharedProperties.searchArray(actions, $scope.id);
+                    for (var i = 0; i < actionIndex; i++) {
+                        var moduleType = filterByUid($scope.moduleData, actions[i].type)[0];
+                        if (moduleType && moduleType.outputs && moduleType.outputs.length) {
+                            $scope.items.push(moduleType.outputs);
```

Comment:
Sorry - `push` will add another array element to an existing array.
![image](https://user-images.githubusercontent.com/2270505/36418915-ae6e72fe-1630-11e8-837c-da05e81909e0.png)

The spread operator would work best here:
```js
$scope.items = [
    ...$scope.items,
    ...moduleType.outputs
]
```

But the question is whether the environment allows you to write in latest and greatest JS possible :)
It would need Babel in the build chain for the backwards compatibility.

Reply:
hell yes, thanks again @kubawolanin!
ES5 is default right now so no cutting edge JS style allowed right now.
For babel, standardjs and further tools we need a CQ before we can actively use it. If you have any suggestions feel free to post them. In parallel to the Tern IDE I will evaluate VSCode as a tool for Paper UI development.

Reply:
reverted.